lunatik.h

```diff
@@ -138,6 +138,15 @@ static inline void *lunatik_checknull(lua_State *L, void *ptr)
 
 #define lunatik_checkalloc(L, s)	(lunatik_checknull((L), lunatik_malloc((L), (s))))
 
+static inline void lunatik_checkfield(lua_State *L, int idx, const char *field, int type)
+{
+	int tname;
```

Comment:
`tname` is a bit misnomer.. it's not the type name that you are storing but the type enum itself